### PR TITLE
Fix: Inconsistent behavior of comma in BigQuery vs TinyQuery (#2501)

### LINKED ISSUE #2501: Inconsistent behavior of comma in BigQuery vs TinyQuery
TinyQuery uses legacy SQL, making some queries incompatible between our unit tests and deployed code. Usually these will manifest as errors but there is an inconsistency with the comma operator that may silently undermine the correctness of the tests.

https://cloud.google.com/bigquery/docs/reference/legacy-sql#comma-as-union-all

In standard SQL the comma is the CROSS JOIN operator but in legacy SQL it is the UNION ALL operator.  A cross join can be unambiguously and consistently expressed in either dialect by writing `CROSS JOIN`, but this leaves us in the brittle position of ensuring we never use comma, and there is no likewise solution for `UNION ALL`.

### PR DESCRIPTION
**Pull request by nadove-ucsc**
https://github.com/DataBiosphere/azul/issues/2501

### Author

- [x] PR title references issue
- [x] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why they're different</sub>
- [x] Title of main commit references issue
- [x] PR is connected to Zenhub issue and description links to issue

### Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

### Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [ ] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [ ] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [ ] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [ ] Decided if PR can be labeled `no sandbox`
- [ ] PR title is appropriate as title of merge commit
- [ ] Moved ticket to Approved column
- [ ] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [ ] Checked `reindex` label and `r` commit title tag
- [ ] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [ ] Rebased and squashed branch
- [ ] Sanity-checked history
- [ ] Pushed PR branch to Github
- [ ] Branch pushed to Gitlab and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [ ] Build passed in sandbox                               <sub>or PR is labeled `no sandbox`</sub>
- [ ] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Added PR reference to merge commit title
- [ ] Collected commit title tags in merge commit title
- [ ] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                 <sub>or PR description does not contain an announcement</sub>
- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate       <sub>or this PR is authored by lead</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab

### Operator (reindex) 

- [ ] Started reindex in target deployment                  <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures             <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment              <sub>or this PR does not require reindexing</sub>

### Operator

- [ ] Unassigned PR

### CODE REVIEW

**Review by dsotirho-ucsc — APPROVED** [review]
Approved with comments for consideration:
- The method name `_query` seems odd to me. I can't find a guideline for it however I think the standard is to name a method by what it does or what it returns. It is of course clear what `_query` does from the type hinting, though I'd still suggest something like `_run_query` instead.
- Can `_union_query` be used in `TestTDRPlugin`, or is that covered by another ticket?

**Comment by noah-aviel-dove** [comment]
I've renamed the methods as you suggested, @danielsotirhos. I'm not sure what you mean by your second question. The motivation behind these additions is so the methods can be used in https://github.com/DataBiosphere/azul/pull/2346, if that's what you're asking.

**Comment by noah-aviel-dove** [comment]
@danielsotirhos I know you already approves but I wanted to follow up on your feedback. Build has passed but is red from coveralls breakage.

File: src/azul/plugins/repository/tdr/__init__.py
Comment (danielsotirhos) [inline src/azul/plugins/repository/tdr/__init__.py:156]:
```suggestion
        return self._run_query(' UNION ALL '.join(subqueries))
```
Currently all queries start and end with whitespace however I don't think `_run_union_query` should assume it to be so.

**Review by dsotirho-ucsc — DISMISSED** [review]
One last suggestion to consider, otherwise looks great.

**Comment by danielsotirhos** [comment]
> I'm not sure what you mean by your second question. The motivation behind these additions is so the methods can be used in #2346, if that's what you're asking.

Yes, that is what I was looking for. Should this PR be based on #2346?

**Comment by noah-aviel-dove** [comment]
> Should this PR be based on #2346?

No, Hannes explicitly requested that this PR be merged first. Once it's approved I'll rebase #2346 on this branch.

**Comment by hannes-ucsc** [comment]
Closing this for now so that it doesn't show up in my queue. I am code owner so I can't cancel the review request.

**Pull request closed without merging**
